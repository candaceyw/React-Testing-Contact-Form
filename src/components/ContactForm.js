import React, { useState } from "react";
import { useForm } from "react-hook-form";

const ContactForm = () => {
  const [data, setData] = useState();
  const { register, errors, handleSubmit, reset } = useForm({
    mode: "onBlur"
  });
  const onSubmit = data => {
    setData(data);
  };



  return (
    <div className="App">
      <form onSubmit={handleSubmit(onSubmit)}>
        <div>
          <label >First Name*
          <input
            name="firstName"
            placeholder="First Name"
            ref={register({ required: true, maxLength: 20 })}
          />
          {errors.firstName && (
            <p>Looks like there was an error: {errors.firstName.type}</p>
          )}
          </label>
        </div>

        <div>
          <label>Last Name*
          <input
            name="lastName"
            placeholder="Last Name"
            ref={register({ required: true })}
          />
          {errors.lastName && (
            <p>Looks like there was an error: {errors.lastName.type}</p>
          )}
          </label>
        </div>

        <div>
          <label >
            Email*
          <input name="email" placeholder="Email" ref={register({ required: true, pattern: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i })} />
          {errors.email && (
            <p>Looks like there was an error: {errors.email.type}</p>
          )}
           </label>
        </div>

        <label>Your prefer pronoun*
        <select name="pronoun" ref={register({ required: true })} >
        <option value="select" defaultValue disabled>select option</option>
        <option value="he/him">he/him</option>
        <option value="she/her">she/her</option>
        <option value="they/them">they/them</option>

      </select>
      </label>


        <div>
          <label>Message
          <textarea name="message" placeholder="Message" ref={register({ required: false })} />
          </label>
        </div>
        {data && (
          <pre style={{ textAlign: "left", color: "white" }}>
            {JSON.stringify(data, null, 2)}
          </pre>
        )}
        <input type="submit" />
      </form>
    </div>
  );
};

export default ContactForm;
