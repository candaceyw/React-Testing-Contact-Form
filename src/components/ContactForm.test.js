import React from 'react'
import ContactForm from './ContactForm'
import { render, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer';

import 'mutationobserver-shim';

test('Renders Label text', () => {

    const { getByLabelText } = render(<ContactForm />);

    const firstName = getByLabelText(/first name/i);
    const lastName = getByLabelText(/last name/i);
    const email = getByLabelText(/email/i);
    const message = getByLabelText(/message/i);

    fireEvent.change(firstName, {target: {value: 'test'}});
    fireEvent.change(lastName, {target: {value: 'test'}})
    fireEvent.change(email, {target: {value: 'test@email.com'}})
    fireEvent.change(message, {target: {value: 'test message'}})

   
    expect(firstName.value).toBe('test');
    expect(lastName.value).toBe('test');
    expect(email.value).toBe('test@email.com');
    expect(message.value).toBe('test message');
    expect('').not.toHaveLength(4);


});


test('renders correct Placeholder text', () => {

    const { getByPlaceholderText } = render(<ContactForm />);

    getByPlaceholderText(/first name/i);
    getByPlaceholderText(/last name/i);
    getByPlaceholderText(/email/i);
    getByPlaceholderText(/Message/i);
})

   /* Email
          @param {string} email
        */
       // Expected: string
       // Match: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i


// render a component with the new renderer, transform it into JSON, and match the snapshot to the previously stored snapshot
describe('ContactForm', () => {
    test('snapshot renders', () => {
      const component = renderer.create(<ContactForm />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });


// test('Checks if field value matches requirement', () => { 
//     const { getByLabelText} = render(<ContactForm />);

//     const firstNameInput = getByLabelText(/first name/i);
//     // const lastNameInput =getByLabelText(/last name/i);
//     // const emailInput =getByLabelText(/email/i);
//     // const messageInput =getByLabelText(/message/i);

//     fireEvent.change(firstNameInput, {target: {value: "Test"}})
//     expect(firstNameInput).toHaveLength(4)


// })



// Write a test that checks if first name is equal or greater than 3
// Check that all fields are required
// Check for errors on all fields