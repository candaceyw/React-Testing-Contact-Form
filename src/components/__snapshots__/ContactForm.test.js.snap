// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm snapshot renders 1`] = `
<div
  className="App"
>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label>
        First Name*
        <input
          name="firstName"
          placeholder="First Name"
        />
      </label>
    </div>
    <div>
      <label>
        Last Name*
        <input
          name="lastName"
          placeholder="Last Name"
        />
      </label>
    </div>
    <div>
      <label>
        Email*
        <input
          name="email"
          placeholder="Email"
        />
      </label>
    </div>
    <label>
      Your prefer pronoun*
      <select
        name="pronoun"
      >
        <option
          defaultValue={true}
          disabled={true}
          value="select"
        >
          select option
        </option>
        <option
          value="he/him"
        >
          he/him
        </option>
        <option
          value="she/her"
        >
          she/her
        </option>
        <option
          value="they/them"
        >
          they/them
        </option>
      </select>
    </label>
    <div>
      <label>
        Message
        <textarea
          name="message"
          placeholder="Message"
        />
      </label>
    </div>
    <input
      type="submit"
    />
  </form>
</div>
`;
